pkg/collector/corechecks/containers/docker/check.go

```diff
@@ -198,25 +201,20 @@ func (d *DockerCheck) runDockerCustom(sender aggregator.Sender, du docker.Client
 		isContainerRunning := rawContainer.State == containers.ContainerRunningState
 		taggerEntityID := containers.BuildTaggerEntityName(rawContainer.ID)
 
-		// Compute container-images stats
-		perImage := containersPerImage[rawContainer.ImageID]
-		if perImage == nil {
-			imageTags, err := getImageTagsFromContainer(taggerEntityID, resolvedImageName, isContainerExcluded || !isContainerRunning)
-			if err != nil {
-				log.Warnf("Unable to fetch tags for container: %s, err: %v", rawContainer.ImageID, err)
-			} else {
-				perImage = &containerPerImage{
-					tags: imageTags,
-				}
-				containersPerImage[rawContainer.ImageID] = perImage
+		tags, err := getImageTagsFromContainer(taggerEntityID, resolvedImageName, isContainerExcluded || !isContainerRunning)
+		if err != nil {
+			log.Warnf("Unable to fetch tags for container: %s, err: %v", rawContainer.ImageID, err)
+		} else {
+			sort.Strings(tags)
+			key := strings.Join(tags, "|")
```

Comment:
This is hot path, maybe we should consider hashing the tag list instead. It's difficult to compare without any benchmarks though.
It shouldn't be a blocker, I know this fix is important for 7.39 so feel free to add a TODO for it and we can tackle it later.

Reply:
Yeah there's a small CPU/Memory tradeoff here, worth benchmarking it

Reply:
Good point. In this case I chose to do the same that we were already doing in a previous release when this was working to avoid introducing to many changes before RC1: https://github.com/DataDog/datadog-agent/blob/7.34.0/pkg/collector/corechecks/containers/docker/docker.go#L90

If you agree, I'll add a note in the QA testing instructions so we remember to take a look to see if this is something that we need to change.

Reply:
I'm not expecting to introduce a perf regression tbh but maybe a card for later to do the benchmarks is better suited here.